### ISSUE TITLE
Datetime format problem

### ISSUE THREAD

**Issue by mrecodes** [apache/incubator-pinot#7195]
I need to import the timestamp from a json file with the format of: YYYY-MM-DDThh:mm:ss.SSSZ for example: 2021-06-22T09:16:31.260Z. The characters T and Z makes problem while trying to visualize with superset. So I tried some transform functions such as:
FromDateTime('timestamp', 'yyyy-MM-dd''T''HH:mm:ss.SSS''Z''')
DATETIMECONVERT(timestamp, '1:MILLISECONDS:SIMPLE_DATE_FORMAT:yyyy-MM-dd''T''HH:mm:ss.SSS''Z', '1:SECONDS:SIMPLE_DATE_FORMAT:yyyy-MM-dd HH:mm:ss', '1:SECONDS')
And many more options, but no luck.
What can be the problem here?

**Comment by xiangfu0** [comment]
I think you can set the time format in  superset, please check this doc: 
https://docs.pinot.apache.org/integrations/superset#adding-a-derived-column

**Comment by mrecodes** [comment]
Yes exactly, I tried %Y-%m-%dT%H:%M:%S.%f and also %Y-%m-%d''T''%H:%M:%S.%f''Z but no luck. From that configuration Superset is trying to run the query that I mentioned in my previous message:
FromDateTime('timestamp', 'yyyy-MM-dd''T''HH:mm:ss.SSS''Z''')
DATETIMECONVERT(timestamp, '1:MILLISECONDS:SIMPLE_DATE_FORMAT:yyyy-MM-dd''T''HH:mm:ss.SSS''Z', '1:SECONDS:SIMPLE_DATE_FORMAT:yyyy-MM-dd HH:mm:ss', '1:SECONDS')
and I get the exception.

**Comment by xiangfu0** [comment]
for pinot queries, single quote is used for literals, double quote is used for identifiers. so 'timestamp ' should be "timestamp".

**Comment by xiangfu0** [comment]
For superset side, have you tried the time format like 
%Y-%m-%dT%H:%M:%S.%fZ or %Y-%m-%dT%H:%M:%SZ
And what's the exceptions?
